pkgs/development/compilers/go/1.21.nix

```diff
@@ -116,14 +110,16 @@ stdenv.mkDerivation rec {
 
   GOROOT_BOOTSTRAP = if useGccGoBootstrap then goBootstrap else "${goBootstrap}/share/go";
 
+  # Note that we use distpack to avoid moving around cross-compiled binaries.
+  # The paths are slightly different when buildPlatform != hostPlatform and
+  # distpack handles assembling outputs in the right place, same as the official
+  # Go binary releases. See also https://pkg.go.dev/cmd/distpack
   buildPhase = ''
     runHook preBuild
     export GOCACHE=$TMPDIR/go-cache
     # this is compiled into the binary
     export GOROOT_FINAL=$out/share/go
 
-    export PATH=$(pwd)/bin:$PATH
```

Comment:
FWIW I think this was added to silence the part of the `cmd/dist` banner that asks to add Go to the `PATH`.
https://github.com/golang/go/blob/5eb382fc08fb32592e9585f9cb99005696a38b49/src/cmd/dist/build.go#L1877-L1879

This is not needed with `-no-banner`.